internal/pkg/admission/policy-server-deployment_test.go

```diff
@@ -210,38 +211,247 @@ func TestIfPolicyServerImageChanged(t *testing.T) {
+		t.Error("Policy server container Capabilities diverge from the expected value")
+	}
+}
+
+func TestPolicyServerWithPodSecurityContext(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+	}
+	var user, group int64
+	user = 1000
+	group = 2000
+	podSecurity := corev1.PodSecurityContext{
+		RunAsUser:  &user,
+		RunAsGroup: &group,
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			SecurityContexts: policiesv1.PolicyServerSecurity{
+				Pod: &podSecurity,
+			},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+
+	if deployment.Spec.Template.Spec.SecurityContext == nil {
+		t.Error("Pod securityContext should be defined ")
+	}
+
+	if deployment.Spec.Template.Spec.SecurityContext != nil {
+		if *deployment.Spec.Template.Spec.SecurityContext.RunAsUser != user {
+			t.Error("Pod RunAsUser diverges from the expected value")
+		}
+		if *deployment.Spec.Template.Spec.SecurityContext.RunAsGroup != group {
+			t.Error("Pod RunAsGroup diverges from the expected value")
+		}
+	}
+}
+
+func TestPolicyServerWithoutSecurityContext(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image:            "image",
+			SecurityContexts: policiesv1.PolicyServerSecurity{},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem != true {
+		t.Error("Policy server container ReadOnlyRootFilesystem diverge from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.Privileged != false {
+		t.Error("Policy server container Privileged diverge from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot != true {
+		t.Error("Policy server container RunAsNonRoot diverges from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation != false {
+		t.Error("Policy server container AllowPrivilegeEscalation diverge from the expected value")
+	}
+	if deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities == nil {
+		t.Error("Policy server container should have capabilities defined")
+	} else {
+		if len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Add) > 0 {
+			t.Error("Policy server container should not have 'Add' capabilities defined")
+		}
+		if len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop) != 1 ||
+			deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop[0] != dropCapabilityAll {
+			t.Error("Policy server container Capabilities should have only one 'All' drop capability")
+		}
+	}
```

Comment:
Here we should compare the actual values with the expected ones, where the expected ones are returned by the `defaultPodSecurityContext` function (if I remember the name of the function properly)

Reply:
Done